### PR TITLE
PPS Pixel DQM: assign errors to right plots

### PR DESCRIPTION
**Pull request by fabferro**
#### PR description:

This PR is a bug fix in filling the plots in PPS Pixel DQM. Using the DAQ map, errors coming from pixels are assigned to the right detector plane instead of being put in a unidentified-detector plot, unless strictly necessary.

#### PR validation:

With matrix wfs 138.4 (run3) and 136.874 (run2) and with brand new 2023 data. Errors are now put inside the right plots.

This fix is needed for the data taking asap. Backports are therefore needed.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by fabferro** [comment]
It looks like auto:run2_mc GT (ie: 123X_mcRun2_asymptotic_v1) does not contain CTPPSPixelDAQMappingRcd. It is needed in the updates introduced by this PR. It should be added. @wpcarvalho

**Comment by emanueleusai** [comment]
type ctpps

**Comment by francescobrivio** [comment]
Hi @fabferro 

the `CTPPSPixelDAQMappingRcd` were introduced in the MC GTs only for 2018 and Run3, as detailed in https://github.com/cms-sw/cmssw/pull/30890 (from 2020) and specifically requested in https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4315/1.html by PPS experts.

So something must be wrong in the PR if you are trying to consume that record in workflows older than 2018...maybe you need to restrict this to Run3 workflows only?

**Comment by francescobrivio** [comment]
Hi @fabferro I think we have multiple possible solutions:
 - either use era modifier
 - add the needed Record to the `auto:run2_mc`
 - But before this I would like to understand why so far we run with no issues. 
    - From the [log summary](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0814e4/32604/runTheMatrix-results/runall-report-step123-.log) I see that only 3 wfs are failing `25202.0, 1306.0, 1330.0` and all three of them are using `--conditions auto:run2_mc`

**Comment by tvami** [comment]
> But before this I would like to understand why so far we run with no issues.

bc the record was not consumed in the past, one of the new lines is:

```
tokenPixelDAQMapping = esConsumes<CTPPSPixelDAQMapping, CTPPSPixelDAQMappingRcd>();
```

so this makes sense. And I think the cleanest way to move on is to include the record in all (relevant) GTs.

**Comment by francescobrivio** [comment]
> > But before this I would like to understand why so far we run with no issues.
> 
> bc the record was not consumed in the past, one of the new lines is:
> 
> ```
> tokenPixelDAQMapping = esConsumes<CTPPSPixelDAQMapping, CTPPSPixelDAQMappingRcd>();
> ```
> 
> so this makes sense. And I think the cleanest way to move on is to include the record in all (relevant) GTs.

Thanks Tamas! So we need the corret payloads to be used in the GT and to make new GTs :) @wpcarvalho I guess it's up to you to provide the relevant payloads and we will take care of making the GTs.

File: DQM/CTPPS/plugins/CTPPSPixelDQMSource.cc [hunk 1]
```diff
@@ -689,6 +697,9 @@ void CTPPSPixelDQMSource::analyze(edm::Event const &event, edm::EventSetup const
   Handle<edm::TriggerResults> hltResults;
   event.getByToken(tokenTrigResults, hltResults);
 
+  ESHandle<CTPPSPixelDAQMapping> mapping;
+  mapping = eventSetup.getHandle(tokenPixelDAQMapping);
```
[the remark this answers is not in the document]
Comment (fabferro) [reply DQM/CTPPS/plugins/CTPPSPixelDQMSource.cc:701]: do you mean eventSetup.getData?

[the remark this answers is not in the document]
Comment (mmusich) [reply DQM/CTPPS/plugins/CTPPSPixelDQMSource.cc:701]: Yes, see above

**Comment by fabferro** [comment]
Reasoning with @wpcarvalho we recalled that, actually, the current PPS detectors were not even present in 2016.
Thus, probably, the best option is to add a few lines in ctppsDQM_cff.py to prevent DQM to be called for 2016 data/mc processes.
Like:
from Configuration.Eras.Modifier_ctpps_2016_cff import ctpps_2016
ctpps_2016.toReplaceWith(
    _ctppsDQMOfflineSource,
    cms.Sequence(
    )   
)

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by emanueleusai** [comment]
+1

**Comment by fabferro** [comment]
> > Reasoning with @wpcarvalho we recalled that, actually, the current PPS detectors were not even present in 2016. Thus, probably, the best option is to add a few lines in ctppsDQM_cff.py to prevent DQM to be called for 2016 data/mc processes. Like: from Configuration.Eras.Modifier_ctpps_2016_cff import ctpps_2016 ctpps_2016.toReplaceWith( _ctppsDQMOfflineSource, cms.Sequence( ) )
> 
> @fabferro I am puzzled: in [#41649 (comment)](https://github.com/cms-sw/cmssw/pull/41649#issuecomment-1547711204) is written that "the CTPPSPixelDAQMappingRcd were introduced in the MC GTs only for 2018 and Run3". Ok for clearing the ctppsDQMOfflineSource from runs in the `Run2_2016` era, but how can this even work for a `Run2_2017` era then?

Actually the campaign for updating PPS MC tags started from 2017, at least for pixels. I'm not 100% sure that ALL MC GTs were updated, but for all used 131X_mc2017_* were.

**Comment by perrotta** [comment]
> Actually the campaign for updating PPS MC tags started from 2017, at least for pixels. I'm not 100% sure that ALL MC GTs were updated, but all used 131X_mc2017_* were.

Ok, thank you @fabferro : this explains why we got no errors from the 2017 workflows

**Comment by perrotta** [comment]
+1

**Pull request merged**